x-pack/platform/plugins/shared/fleet/server/services/secrets.ts

```diff
@@ -266,7 +266,9 @@ export async function extractAndWriteSecrets(opts: {
     return { packagePolicy, secretReferences: [] };
   }
 
-  const secretsToCreate = secretPaths.filter((secretPath) => !!secretPath.value.value);
+  const secretsToCreate = secretPaths.filter(
```

Comment:
I'm having a hard time following the logic of secrets extraction in this file.

can you give some snippets of what the expected input & output is like for a package policy with and without a cloud connector?

also, the changes here need tests (that might help me understand this better too)